### PR TITLE
Sync/upstream main 20260712

### PR DESCRIPTION
**Pull request by ralampay**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Adds custom detection backbones and attention-based convolution modules, including a DraxNet integration for YOLO26, alongside dataset utilities, model configurations, tests, and upstream synchronization tooling.

### 📊 Key Changes
- Adds DraxNet, Drax residual blocks, ConvNeXt-style mixing, spatial attention, stochastic depth, and multi-scale feature outputs.
- Adds custom modules such as `ConvAttnLite`, `CoordAttConv`, `MyConvBlock`, and optional deformable convolution support.
- Registers the new modules with Ultralytics YAML model parsing and exports them through the neural network module package.
- Adds the `draxnet-yolo26.yaml`, custom YOLO11, and custom YOLO12 model configurations.
- Adds construction and forward-pass tests for DraxNet models and feature shapes.
- Adds a COCO-to-person/vehicle/animal conversion script and upstream merge helper documentation/scripts.
- Adds scratch and notes files, plus a standalone custom model smoke-test script.

### 🎯 Purpose & Impact
- Enables experimentation with non-standard backbones and attention modules while retaining the existing Ultralytics detection pipeline.
- Provides multi-scale DraxNet features compatible with YOLO26 detection heads.
- Improves extensibility for custom YAML-defined architectures and introduces basic regression coverage.
- Includes several repository-level utilities, but the broad scope and direct modification of the shared YOLO12 class count require careful review to avoid unintended behavior changes.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ralampay, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your custom backbones and detection modules, please review the following checklist:
- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). This is an automated message, and an engineer will assist with any questions or review follow-up. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/cfg/models/12/yolo12.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Task docs: https://docs.ultralytics.com/tasks/detect
 
 # Parameters
-nc: 80 # number of classes
+nc: 1 # number of classes
```
Comment (UltralyticsAssistant) [inline ultralytics/cfg/models/12/yolo12.yaml:8]: ⚠️ **HIGH**: This changes the canonical YOLO12 configuration from the standard 80 COCO classes to a single class. Direct users of `YOLO("yolo12.yaml")`/`yolo12n.yaml` will now construct a one-class detection head, and the checked-in parameter/FLOPs summaries no longer match. Keep the upstream model at `nc: 80` and put the one-class setting only in a separate custom dataset/model YAML.

**Suggested change:**
```suggestion
nc: 80 # number of classes
```

File: bin/merge.sh [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+if (( has_tracked_changes || has_untracked_files )); then
+  echo "Committing current changes on $current_branch..."
+  git add -A
+  git commit -m "$commit_message"
+fi
+
+echo "Fetching $remote..."
+git fetch --prune --tags "$remote"
+
+latest_release_tag="$(git tag --list 'v*' --sort=-version:refname | head -n 1)"
```
Comment (UltralyticsAssistant) [inline bin/merge.sh:60]: 💡 **MEDIUM**: `git tag --list` reads every local tag, including stale tags and tags created locally that did not come from `$remote`. A higher local `v*` tag can therefore be selected and rebased onto even though it is not the latest upstream release. Resolve the release tag from the remote refs (or constrain the selection to tags verified as fetched from `$remote`) before rebasing.

File: test_custom.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from ultralytics import YOLO
+import gc
+import torch
+
+gc.collect()
+torch.cuda.empty_cache()
+
+model = YOLO("ultralytics/cfg/models/ext/cad_yolo12.yaml")
```
Comment (UltralyticsAssistant) [inline test_custom.py:8]: 💡 **MEDIUM**: This smoke test references `cad_yolo12.yaml`, but the added configuration is named `cal_yolo12.yaml`. Running the script currently fails while resolving the model path.

**Suggested change:**
```suggestion
model = YOLO("ultralytics/cfg/models/ext/cal_yolo12.yaml")
```

File: ultralytics/nn/modules/custom_blocks.py [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        kernel_size = tuple(kernel)
+
+        stride = (s, s) if isinstance(s, int) else tuple(s)
+        if isinstance(d, int):
+            dilation = (d, d)
+            d_for_pad = d
+        else:
+            dilation = tuple(d)
+            d_for_pad = list(d)
+        padding = autopad(kernel, p, d_for_pad)
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/modules/custom_blocks.py:71]: 💡 **MEDIUM**: The declared `Sequence[int]` dilation API is not usable here: when `d` is a tuple/list, `d_for_pad` is passed to `autopad()`, whose `if d > 1` comparison raises `TypeError` for a sequence. The same pattern appears in `ConvAttnDeform`; normalize dilation/padding locally or restrict `d` to an integer.

File: extended_backbones.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# Extended Backbones
+
+This repository can expose non-standard backbones to Ultralytics YAMLs as native modules. The `draxnet-yolo26` example added here shows the pattern for adapting an external backbone into a YOLO detection model without changing the rest of the training and inference stack.
+
+## What Was Added
+
+- `DraxBlock` was copied from the sibling MLX project and added to [ultralytics/nn/modules/block.py](/home/ralampay/workspace/ultralytics/ultralytics/nn/modules/block.py).
```
Comment (UltralyticsAssistant) [inline extended_backbones.md:7]: 📝 **LOW**: These are absolute paths from a local workstation, so the links are broken for every repository user and expose machine-specific filesystem details. Use repository-relative links instead.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The DraxNet parser integration and targeted construction tests are a solid start, but the canonical YOLO12 class-count regression, incorrect smoke-test path, and remote-tag selection need correction. There is also an advertised custom-block API path that currently crashes and documentation with non-portable links.

💬 Posted 5 inline comments
